hub/elastic_sync/service.py

```diff
@@ -174,35 +176,31 @@ async def apply_filters(self, blocked_streams, blocked_channels, filtered_stream
         if filtered_streams:
             await self.sync_client.update_by_query(
                 self.index, body=self.update_filter_query(Censor.SEARCH, only_channels(filtered_streams)), slices=4)
-            await self.sync_client.indices.refresh(self.index)
         if filtered_channels:
             await self.sync_client.update_by_query(
                 self.index, body=self.update_filter_query(Censor.SEARCH, only_channels(filtered_channels)), slices=4)
-            await self.sync_client.indices.refresh(self.index)
             await self.sync_client.update_by_query(
                 self.index, body=self.update_filter_query(Censor.SEARCH, only_channels(filtered_channels), True), slices=4)
-            await self.sync_client.indices.refresh(self.index)
         if blocked_streams:
             await self.sync_client.update_by_query(
                 self.index, body=self.update_filter_query(Censor.RESOLVE, only_channels(blocked_streams)), slices=4)
-            await self.sync_client.indices.refresh(self.index)
         if blocked_channels:
             await self.sync_client.update_by_query(
                 self.index, body=self.update_filter_query(Censor.RESOLVE, only_channels(blocked_channels)), slices=4)
-            await self.sync_client.indices.refresh(self.index)
             await self.sync_client.update_by_query(
                 self.index, body=self.update_filter_query(Censor.RESOLVE, only_channels(blocked_channels), True), slices=4)
+        if filtered_streams or filtered_channels or blocked_streams or blocked_channels:
             await self.sync_client.indices.refresh(self.index)
 
     @staticmethod
     def _upsert_claim_query(index, claim):
-        return {
-            'doc': {key: value for key, value in claim.items() if key in ALL_FIELDS},
+        doc = {key: value for key, value in claim.items() if key in ALL_FIELDS}
+        doc.update({
             '_id': claim['claim_id'],
             '_index': index,
-            '_op_type': 'update',
-            'doc_as_upsert': True
-        }
+            '_op_type': 'index',
+        })
+        return doc
```

Comment:
This surprised me, but it appears that the old way of doing this did not handle deletions of extensions. It's like it was applying `doc` as a set of changes to the existing doc.

The new way is intended to completely replace the `doc` with new contents.

Reply:
It appears this was not necessary. I must have had some other problem involving the `ALL_FIELDS` or `FIELDS` definitions or the `prepare_claim_metadata_batch() definition`. Reverted.